cloudinit/distros/rhel.py

```diff
@@ -94,6 +94,10 @@ def _write_hostname(self, hostname, out_fn):
     def _select_hostname(self, hostname, fqdn):
         # Should be fqdn if we can use it
         # See: https://www.centos.org/docs/5/html/Deployment_Guide-en-US/ch-sysconfig.html#s2-sysconfig-network # noqa
+        if not util.get_cfg_option_bool(self._cfg,
```

Comment:
This area of the code is tested with  tests/unittests/test_handler/test_handler_set_hostname.py  as well. So I suppose exending those tests with a new test to exercise  this logic  would be a gate to landing this.